{"text": "2023-10-27 07:20:39 | INFO    | __main__:stash:129 | run - {'image': '20231026', 'role': 'EpicPlayer', 'headless': True, 'self_supervised': True}\n", "record": {"elapsed": {"repr": "0:00:01.915715", "seconds": 1.915715}, "exception": null, "extra": {"image": "20231026", "role": "EpicPlayer", "headless": true, "self_supervised": true}, "file": {"name": "claim.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/claim.py"}, "function": "stash", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 129, "message": "run", "module": "claim", "name": "__main__", "process": {"id": 3558, "name": "MainProcess"}, "thread": {"id": 139874655538048, "name": "MainThread"}, "time": {"repr": "2023-10-27 07:20:39.323482+08:00", "timestamp": 1698362439.323482}}}
{"text": "2023-10-27 07:20:52 | INFO    | __main__:claim_epic_games:95 | Try to flush cookie - {'task': 'claim_epic_games'}\n", "record": {"elapsed": {"repr": "0:00:14.702849", "seconds": 14.702849}, "exception": null, "extra": {"task": "claim_epic_games"}, "file": {"name": "claim.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/claim.py"}, "function": "claim_epic_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 95, "message": "Try to flush cookie", "module": "claim", "name": "__main__", "process": {"id": 3558, "name": "MainProcess"}, "thread": {"id": 139874655538048, "name": "MainThread"}, "time": {"repr": "2023-10-27 07:20:52.110616+08:00", "timestamp": 1698362452.110616}}}
{"text": "2023-10-27 07:20:55 | SUCCESS    | epic_games.agent:_login:203 | login - {'result': 'Successfully refreshed tokens'}\n", "record": {"elapsed": {"repr": "0:00:18.380684", "seconds": 18.380684}, "exception": null, "extra": {"result": "Successfully refreshed tokens"}, "file": {"name": "agent.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py"}, "function": "_login", "level": {"icon": "‚úÖ", "name": "SUCCESS", "no": 25}, "line": 203, "message": "login", "module": "agent", "name": "epic_games.agent", "process": {"id": 3558, "name": "MainProcess"}, "thread": {"id": 139874655538048, "name": "MainThread"}, "time": {"repr": "2023-10-27 07:20:55.788451+08:00", "timestamp": 1698362455.788451}}}
{"text": "2023-10-27 07:21:09 | SUCCESS    | epic_games.agent:flush_token:224 | flush_token - {'path': PosixPath('/home/runner/work/baipiao/baipiao/epic/user_data_dir/epic-games@934175043/ctx_cookie.json')}\n", "record": {"elapsed": {"repr": "0:00:32.463754", "seconds": 32.463754}, "exception": null, "extra": {"path": "/home/runner/work/baipiao/baipiao/epic/user_data_dir/epic-games@934175043/ctx_cookie.json"}, "file": {"name": "agent.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py"}, "function": "flush_token", "level": {"icon": "‚úÖ", "name": "SUCCESS", "no": 25}, "line": 224, "message": "flush_token", "module": "agent", "name": "epic_games.agent", "process": {"id": 3558, "name": "MainProcess"}, "thread": {"id": 139874655538048, "name": "MainThread"}, "time": {"repr": "2023-10-27 07:21:09.871521+08:00", "timestamp": 1698362469.871521}}}
{"text": "2023-10-27 07:21:10 | WARNING    | epic_games.agent:get_order_history:389 | Failed to get order history, cookie may have expired - {}\n", "record": {"elapsed": {"repr": "0:00:32.969123", "seconds": 32.969123}, "exception": null, "extra": {}, "file": {"name": "agent.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py"}, "function": "get_order_history", "level": {"icon": "‚ö†Ô∏è", "name": "WARNING", "no": 30}, "line": 389, "message": "Failed to get order history, cookie may have expired", "module": "agent", "name": "epic_games.agent", "process": {"id": 3558, "name": "MainProcess"}, "thread": {"id": 139874655538048, "name": "MainThread"}, "time": {"repr": "2023-10-27 07:21:10.376890+08:00", "timestamp": 1698362470.37689}}}
{"text": "2023-10-27 07:21:10 | DEBUG    | __main__:create_tasks:64 | Put task - {'title': 'The Evil Within 2', 'url': 'https://store.epicgames.com/en-US/p/the-evil-within-2'}\n", "record": {"elapsed": {"repr": "0:00:33.393342", "seconds": 33.393342}, "exception": null, "extra": {"title": "The Evil Within 2", "url": "https://store.epicgames.com/en-US/p/the-evil-within-2"}, "file": {"name": "claim.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 64, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3558, "name": "MainProcess"}, "thread": {"id": 139874655538048, "name": "MainThread"}, "time": {"repr": "2023-10-27 07:21:10.801109+08:00", "timestamp": 1698362470.801109}}}
{"text": "2023-10-27 07:21:10 | INFO    | epic_games.agent:claim_weekly_games:232 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/the-evil-within-2'}\n", "record": {"elapsed": {"repr": "0:00:33.393990", "seconds": 33.39399}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/the-evil-within-2"}, "file": {"name": "agent.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 232, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3558, "name": "MainProcess"}, "thread": {"id": 139874655538048, "name": "MainThread"}, "time": {"repr": "2023-10-27 07:21:10.801757+08:00", "timestamp": 1698362470.801757}}}
{"text": "2023-10-27 07:21:23 | ERROR    | __main__:run:155 | An error has been caught in function 'run', process 'MainProcess' (3558), thread 'MainThread' (139874655538048): - {}\nTraceback (most recent call last):\n  File \"/home/runner/work/baipiao/baipiao/epic/src/claim.py\", line 159, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n> File \"/home/runner/work/baipiao/baipiao/epic/src/claim.py\", line 155, in run\n    await agent.stash()\n  File \"/home/runner/work/baipiao/baipiao/epic/src/claim.py\", line 148, in stash\n    await self.claim_epic_games(context)\n  File \"/home/runner/work/baipiao/baipiao/epic/src/claim.py\", line 120, in claim_epic_games\n    await epic.claim_weekly_games(page, single_promotions)\n  File \"/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py\", line 248, in claim_weekly_games\n    await expect(cta_btn).to_have_text(\"View In Cart\")\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py\", line 20080, in to_have_text\n    await self._impl_obj.to_have_text(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 455, in to_have_text\n    await self._expect_impl(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 70, in _expect_impl\n    raise AssertionError(f\"{out_message}\\nActual value: {actual} {log}\")\nAssertionError: Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\nwaiting for locator(\"xpath=//aside//button[@data-testid='add-to-cart-cta-button']\")\n", "record": {"elapsed": {"repr": "0:00:46.319434", "seconds": 46.319434}, "exception": {"type": "AssertionError", "value": "Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\nwaiting for locator(\"xpath=//aside//button[@data-testid='add-to-cart-cta-button']\")", "traceback": true}, "extra": {}, "file": {"name": "claim.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/claim.py"}, "function": "run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 155, "message": "An error has been caught in function 'run', process 'MainProcess' (3558), thread 'MainThread' (139874655538048):", "module": "claim", "name": "__main__", "process": {"id": 3558, "name": "MainProcess"}, "thread": {"id": 139874655538048, "name": "MainThread"}, "time": {"repr": "2023-10-27 07:21:23.727201+08:00", "timestamp": 1698362483.727201}}}
